src/Copula.cpp

```diff
@@ -0,0 +1,229 @@
+#include <RcppArmadillo.h>
+using namespace Rcpp;
+
+// [[Rcpp::depends(RcppArmadillo)]]
+
+//' Compute the quantiles using quantile type seven
+//'
+//' @details Using quantile type number seven from stats::quantile in R.
+//'
+//' @param x arma::vec. Numeric vector whose sample quantiles are wanted.
+//' @param probs arma::vec. Numeric vector of probabilities with values between zero and one.
+//'
+//' @return A vector of length `length(probs)` with the quantiles is returned.
+//'
+//' @keywords internal
+//' @author Lars Henry Berge Olsen
+// [[Rcpp::export]]
+arma::vec quantile_type7_cpp(const arma::vec& x, const arma::vec& probs) {
+  int n = x.n_elem;
+  int m = probs.n_elem;
+
+  // Initialize output quantile vector
+  arma::vec qs(m);
+
+  // Calculate indices
+  arma::vec index = 1 + (n - 1) * probs;
+  arma::vec lo = arma::floor(index);
+  arma::vec hi = arma::ceil(index);
+
+  // Sort the data
+  arma::vec sorted_x = arma::sort(x);
+
+  // Calculate quantiles using quantile type seven
+  for (int i = 0; i < m; ++i) {
+    qs(i) = sorted_x(lo(i) - 1);
+    if (index(i) > lo(i)) {
+      double h = index(i) - lo(i);
+      qs(i) = (1 - h) * qs(i) + h * sorted_x(hi(i) - 1);
+    }
+  }
+
+  return qs;
+}
+
+//' Transforms new data to a standardized normal distribution
+//'
+//' @details The function uses `arma::quantile(...)` which corresponds to R's `stats::quantile(..., type = 5)`.
```

Comment:
Correct this. Uses type = 7.

Reply:
Removed the whole sentence as we are not longer using arma::quantile()